### ISSUE TITLE
Migrate from JavaScript --> TypeScript

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/tasking-manager#6533]
**Is your feature request related to a problem? Please describe.**

- Currently the codebase is JavaScript.
- It's a bit messy as we do have some TypeScript declarations inside JS files...
- We probably need a big refactor to improve the maintainability of the codebase.
- Other issues such as https://github.com/hotosm/tasking-manager/issues/6531 are partly blocked by this.

**Describe the solution you'd like**

- Rename `.js` files to `.jsx`.
- Upgrade all code JavaScript --> TypeScript, with `js` and `jsx` to `ts` and `tsx`.

**Describe alternatives you've considered**

- JSDoc, but IMO its better for installable packages, not for software tools, where typescript shines.

**Additional context**

- This is low priority, but will be helpful into the future.
- @JoltCode has kindly volunteered to look into this.
- It's a minor blocker for some other tasks, as it makes sense to do this first.

**Comment by spwoodcock** [comment]
For some reason Github is being funny and I can't assign @JoltCode for now - I will update when possible - but for now it's noted here that @JoltCode will be working on this 👍
